### PR TITLE
Apify Dataset: Remove unnecessary breaking change and improve documentation

### PR DESCRIPTION
**Pull request by flash1293**
This PR removes the unnecessary breaking change for 2.1.0 and clarifies the breaking change in 2.0.0

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-apify-dataset test report (commit [`e0f9b4dfa3`](https://github.com/airbytehq/airbyte/pull/31462/commits/e0f9b4dfa30ccfb7a2bc052f82631533d384e3c6)) - ✅

⏲️  Total pipeline duration: 01mn28s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build source-apify-dataset docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                     | ✅       |
| Code format checks                                                   | ✅       |
| Validate metadata for source-apify-dataset                           | ✅       |
| Connector version semver check                                       | ✅       |
| QA checks                                                            | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_cleanup-apify/1697475781/e0f9b4dfa30ccfb7a2bc052f82631533d384e3c6/source-apify-dataset/2.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e0f9b4dfa30ccfb7a2bc052f82631533d384e3c6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-apify-dataset test
```

File: airbyte-integrations/connectors/source-apify-dataset/metadata.yaml [hunk 1]
```diff
@@ -25,10 +25,7 @@ data:
         message: "Update spec to use token and ingest all 3 streams correctly"
       2.0.0:
         upgradeDeadline: 2023-09-18
-        message: "Fix broken stream, manifest refactor"
-      2.1.0:
-        upgradeDeadline: 2023-10-27
-        message: "Rename dataset streams"
+        message: "Rename streams, adjust item collection schema to retain data for Web-Content-Crawler datasets"
```
Comment (pedroslopez) [inline airbyte-integrations/connectors/source-apify-dataset/metadata.yaml:28]:
```suggestion
        message: "This version introduces a new Item Collection (WCC) stream as a substitute of the now-removed Item Collection stream in order to retain data for Web-Content-Crawler datasets.
```

File: docs/integrations/sources/apify-dataset-migrations.md [hunk 2]
```diff
@@ -1,12 +1,8 @@
 # Apify Dataset Migration Guide
 
-## Upgrading to 2.1.0
-
-A minor update adding a new stream `item_collection` for general datasets. No actions are required regarding your current connector configuration setup.
-
 ## Upgrading to 2.0.0
 
-Major update: The old broken Item Collection stream has been removed and replaced with a new Item Collection (WCC) stream specific for the datasets produced by [Website Content Crawler](https://apify.com/apify/website-content-crawler) Actor. Please update your connector configuration setup. Note: The schema of the Apify Dataset is at least Actor-specific, so we cannot have a general Stream with a static schema for getting data from a Dataset.
+Major update: The old broken Item Collection stream has been removed and replaced with a new Item Collection (WCC) stream specific for the datasets produced by [Website Content Crawler](https://apify.com/apify/website-content-crawler) Actor. In a follow-up release 2.1.0, a generic item collection stream will be added to support all other datasets.
```
[the remark this answers is not in the document]
Comment (pedroslopez) [reply docs/integrations/sources/apify-dataset-migrations.md:5]: Also, is that new stream in the follow up released named the same? can existing users migrate directly to it?

**Review by pedroslopez — COMMENTED** [review]
Some suggestions, main thing is I think we should express the breaking change message in terms of user value to encourage the upgrade as this is what they see in-app

**Review by pedroslopez — APPROVED** [review]
Thanks!

**Pull request merged**